From 0af83e561e05f8ce72b4f67057d54c10d1a1c396 Mon Sep 17 00:00:00 2001
From: Alaa Hleihel <ahleihel@redhat.com>
Date: Mon, 1 Jun 2020 15:40:53 -0400
Subject: [PATCH 298/312] [netdrv] net/mlx5e: en_tc: Fix incorrect type in
 initializer warnings

Message-id: <20200601154102.25980-31-ahleihel@redhat.com>
Patchwork-id: 315735
Patchwork-instance: patchwork
O-Subject: [RHEL8.3 BZ 1842258 30/39] net/mlx5e: en_tc: Fix incorrect type in initializer warnings
Bugzilla: 1842258
RH-Acked-by: Honggang Li <honli@redhat.com>
RH-Acked-by: Kamal Heib <kheib@redhat.com>
RH-Acked-by: Marcelo Leitner <mleitner@redhat.com>
RH-Acked-by: Jarod Wilson <jarod@redhat.com>

Bugzilla: http://bugzilla.redhat.com/1842258
Upstream: git://git.kernel.org/pub/scm/linux/kernel/git/davem/net-next.git , branch: master

commit c51323ee7ab4132c80db198b7d0956fef957e6ab
Author: Saeed Mahameed <saeedm@mellanox.com>
Date:   Wed May 27 23:41:03 2020 -0700

    net/mlx5e: en_tc: Fix incorrect type in initializer warnings

    Fix some trivial warnings of the type:
    warning: incorrect type in initializer (different base types)

    Signed-off-by: Saeed Mahameed <saeedm@mellanox.com>
    Reviewed-by: Mark Bloch <markb@mellanox.com>

Signed-off-by: Alaa Hleihel <ahleihel@redhat.com>
Signed-off-by: Frantisek Hrbata <fhrbata@redhat.com>
---
 drivers/net/ethernet/mellanox/mlx5/core/en_tc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/mellanox/mlx5/core/en_tc.c b/drivers/net/ethernet/mellanox/mlx5/core/en_tc.c
index 9ee982366893..5a15e4630171 100644
--- a/drivers/net/ethernet/mellanox/mlx5/core/en_tc.c
+++ b/drivers/net/ethernet/mellanox/mlx5/core/en_tc.c
@@ -1828,7 +1828,7 @@ enc_opts_is_dont_care_or_full_match(struct mlx5e_priv *priv,
 		    memchr_inv(opt->opt_data, 0, opt->length * 4)) {
 			*dont_care = false;
 
-			if (opt->opt_class != U16_MAX ||
+			if (opt->opt_class != htons(U16_MAX) ||
 			    opt->type != U8_MAX) {
 				NL_SET_ERR_MSG(extack,
 					       "Partial match of tunnel options in chain > 0 isn't supported");
-- 
2.13.6

